scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -443,7 +443,8 @@ def solve(b):
         return splu(A).solve
 
 
-def spsolve_triangular(A, b, lower=True, overwrite_A=False, overwrite_b=False):
+def spsolve_triangular(A, b, lower=True, unit_diagonal=False,
+                       overwrite_A=False, overwrite_b=False):
```

Comment:
What happens to old code that was calling with purely positional arguments:

`spsolve_triangular(A, b, True, True, True)`

Doesn't this change the meaning silently to have `unit_diagonal` set to `True`, and `overwrite_b` adjusted from `True` to default `False`?

Reply:
Yes, you are right. I had used the order of `scipy.linalg.solve_triangular` so that both signatures are more similar. But this is not backwards-compatibility with purely positional arguments. So I changed this.